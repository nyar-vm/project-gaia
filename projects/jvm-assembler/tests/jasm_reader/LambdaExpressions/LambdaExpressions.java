import java.util.function.Function;
import java.util.function.Consumer;

public class LambdaExpressions {
    private String prefix = "Result: ";
    
    public Function<Integer, String> createLambda() {
        return (Integer x) -> prefix + x * 2;
    }
    
    public void processWithLambda(Consumer<String> processor) {
        processor.accept("Hello Lambda");
    }
    
    public void methodReference() {
        Consumer<String> printer = System.out::println;
        printer.accept("Method Reference");
    }
    
    public void nestedLambda() {
        Function<Integer, Function<Integer, Integer>> adder = x -> y -> x + y;
        int result = adder.apply(5).apply(3);
    }
}