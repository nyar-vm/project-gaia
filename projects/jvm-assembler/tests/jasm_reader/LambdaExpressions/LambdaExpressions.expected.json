{
    "class_name": "LambdaExpressions",
    "class_modifiers": [
        "public",
        "super"
    ],
    "method_count": 2,
    "field_count": 0,
    "expected_methods": [
        "<init>:()V",
        "createLambda:()Ljava/util/function/Function;"
    ],
    "expected_fields": [],
    "syntactic_sugar_features": [],
    "check_tokens": true,
    "source_file": null
}